#### Задание 1.

Из колоды в 52 карты извлекаются случайным образом 4 карты. a) Найти вероятность того, что все карты – крести. б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

In [1]:
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

In [5]:
# a)
c1 = combinations(13, 4) # Вычисление количества сочетаний 4-х карт среди всех крестей
c2 = combinations(52, 4) # Вычисление общего количества сочетаний 4-х карт среди всей колоды
p = c1 / c2
print(c1, c2, p)

715 270725 0.0026410564225690276


Ответ: вероятность того, что все карты крести - 0.264%

In [6]:
# b)

c1 = combinations(4, 1) * combinations(48, 3) # Количество комбинаций с одним тузом
c2 = combinations(4, 2) * combinations(48, 2) # Количество комбинаций с двумя тузами
c3 = combinations(4, 3) * combinations(48, 1) # Количество комбинаций с тремя тузами
c4 = 1 # Количество комбинаций с 4-мя тузами
c = c1 + c2 + c3 + c4
p = c / combinations(52, 4)
print(c1, c2, c3, c, p)

69184 6768 192 76145 0.2812632745405855


Ответ: вероятность, что среди 4-х карт окажется хотя бы один туз - 28.126%

#### Задание 2.

На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

In [7]:
p = 1 / combinations(10, 3)
print(p)

0.008333333333333333


Ответ: если кодовый замок новый, вероятность его открытия с первого раза 0.8% =)

#### Задание 3.

В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

In [8]:
c1 = combinations(9, 3) # Количество сочетаний окрашенных деталей
c2 = combinations(15, 3) # Количество всех комбинаций
p = c1 / c2
print(c1, c2, p)

84 455 0.18461538461538463


Ответ: вероятность вытащить все окрашенные детали - 18.46%

#### Задание 4.
В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

In [9]:
p = 1 / combinations(100, 2)
print(p)

0.00020202020202020202


Ответ: вероятность того, что 2 приобретенных билета окажутся выигрышными - 0.02%